build.sh

```diff
@@ -119,7 +119,7 @@ fi
 USE_JEMALLOC=ON
 
 HELP=0
-if [ $# == 1 ] ; then
+if [[ $OPTS =~ "-j" ]] && [ $# == 3 ] || [ $# == 1 ] ; then
```

Comment:
Don't get why we need the regex here.

Reply:
Thanks for your comment. This make sure that the OPTS array contains `-j`.

Reply:
it is not required to add these code.
Please run 
```
 bash -x ./build.sh -j 12 --clean --be
```
you will see output as follows:
```
time make -j 12
```
this means that -j opt is handled correctly.

Reply:
But when we run sh build.sh -j 12, without other params, will cause the compilation run on empty. We can see this [issue](https://github.com/StarRocks/starrocks/issues/11597).